### PR TITLE
Bundle v2

### PR DESCRIPTION
**Pull request by mikeynap**
Removed BundleServices. Kept Plugins.

This is the first step -- another PR will add 'integrating' the other services outputs into the bundle's main service.

Before reviewing, read up on bundles in the docs (gist: https://gist.github.com/mikeynap/9a0b880633950e69dad50a71958543a2)

### CODE REVIEW

File: api/v1/server/server.go [hunk 1]
```diff
@@ -76,7 +76,11 @@ func NewConfig() Config {
 
 //New creates a new http server instance by taking in a url to listen on and a database object
 func New(cfg Config, db database.Database) (*Server, error) {
-	if _, err := url.Parse(cfg.URL); err != nil {
+	tUrl := cfg.URL
+	if !strings.Contains(cfg.URL, "//") {
+		tUrl = "//" + tUrl
+	}
+	if _, err := url.Parse(tUrl); err != nil {
```
Comment (adamschaub) [inline api/v1/server/server.go]: were we going to remove this?

File: config.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@ Log:
   Level: debug
 Manager:
   OperationUpdateInterval: 10s
-  ServiceStatusInterval: 1m
+  ServiceStatusInterval: 2m
```
Comment (adamschaub) [inline config.yaml]: 1m30s

File: database/couchbase.go [hunk 3]
```diff
@@ -82,18 +81,20 @@ func NewCouchbase(cfg CouchbaseConfig) (*Couchbase, error) {
 		cb.cfg.timeout = t
 	}
 
-	if u, err := url.Parse(cfg.Host); err != nil {
+	h := cfg.Host
+	if !strings.Contains(h, "//") {
+		h = "//" + h
+	}
+	if _, err = url.Parse(h); err != nil {
```
Comment (adamschaub) [inline database/couchbase.go]: Remove?

File: manager/manager_test.go [hunk 4]
```diff
@@ -975,6 +987,98 @@ func TestCollectInput(t *testing.T) {
 			true,
 			false,
 			nil,
+			nil,
+		},
+		// from a BundleService
```
Comment (adamschaub) [inline manager/manager_test.go]: From a what?

File: config.yaml (hunk 2 above)
Comment (mikeynap) [inline config.yaml]: .

File: manager/manager_test.go [hunk 5]
```diff
@@ -975,6 +987,98 @@ func TestCollectInput(t *testing.T) {
...
+					{Name: "from-multiple-bundle-services", From: []string{"bs.out_multi", "svc2.out1"}, Type: "list/string"},
+				},
+			},
+			model.Plugin{},
+			true,
+			true,
+			[]string{"val", "val2", "val1"},
+			nil,
+		},
+		// from recursive BundleService
```
Comment (adamschaub) [inline manager/manager_test.go]: What's that?

File: manager/manager_test.go (hunk 4 above)
Comment (mikeynap) [inline manager/manager_test.go]: short hand for "a service that starts a bundle", I'll clarify it.

File: tests/integration/manager/manager_test.go [hunk 6]
```diff
@@ -117,7 +121,7 @@ func testServices(tt *testing.T) error {
 			map[string][]string{"url": []string{"127.0.1.2"}, "foo-output": []string{"Hello, ", "World"}},
 			map[string][]string{
 				"url":        []string{"127.0.1.2"},
-				"foo-output": []string{"Hello, ", "World"}, // DEPENDS ON ADAM PR. FIX ME.
+				"foo-output": []string{"Hello, ", "World"},
```
Comment (adamschaub) [inline tests/integration/manager/manager_test.go]: Guess it didn't matter?

File: manager/manager.go [hunk 7]
```diff
@@ -151,62 +149,74 @@ func (m *manager) Stop() {
...
-		bs, err := m.db.GetBundleService(bss.Id)
-		if err != nil {
-			log.Errorf("Could not get bundle services (for id %v)", bss.Id)
-			continue
-		}
-		bundle, err := m.bundleFor(bs)
 
+	// populate outputs from plugin
+	dumbyOutput := make(map[string]interface{})
+	for i, svc := range bundle.Services {
```
Comment (adamschaub) [inline manager/manager.go]: Can we not collapse both of these loops?

File: manager/manager.go [hunk 8]
```diff
@@ -151,62 +149,74 @@ func (m *manager) Stop() {
...
-			}
-			svc.Set("bundleservice", bs.Name)
-			svc.Set("bundleservice_id", bs.Id)
+		collectOutput(&bundle.Services[i], plugin, dumbyOutput)
+	}
 
-			if _, err = m.db.InsertService(svc); err != nil {
-				log.WithError(err).Error("Unable to insert service from bundle.")
-			}
+	for _, svc := range bundle.Services {
```
Comment (adamschaub) [inline manager/manager.go]: This and the one above?

File: manager/manager.go [hunk 9]
```diff
@@ -151,62 +149,74 @@ func (m *manager) Stop() {
...
 		// DB Get so CAS is acquired
-		var err error
-		if service, err = m.db.GetService(service.Id); err != nil {
+		var service model.Service
+		if service, err = m.db.GetService(serviceID); err != nil {
 			log.WithError(err).Errorf("Could not get service %q", service.Name)
 			continue
 		}
 
+		if service.IsBundleService() {
```
Comment (adamschaub) [inline manager/manager.go]: I like

File: manager/manager.go (hunk 7 above)
Comment (mikeynap) [inline manager/manager.go]: good question

**Comment by adamschaub** [comment]
Outside of that repeated `for` over the bundle services and the `url.Parse` bit, I'm good to merge

**Comment by adamschaub** [comment]
LOVE IT

**Pull request merged**